service/storage_service.cc

```diff
@@ -2888,6 +2894,40 @@ future<> storage_service::restore_replica_count(inet_address endpoint, inet_addr
         }
         streamer->add_rx_ranges(keyspace_name, std::move(ranges_per_endpoint));
     }
+    auto status_checker = seastar::async([this, endpoint, &as] {
+        slogger.info("restore_replica_count: Started status checker for removing node {}", endpoint);
+        while (!as.abort_requested()) {
+            auto status = _gossiper.get_gossip_status(endpoint);
+            // If the node to be removed is already in removed status, it has
+            // probably been removed forcely with `nodetool removenode force`.
+            // Abort the restore_replica_count in such case to avoid streaming
+            // attempt since the user has removed the node forcely.
+            if (status == sstring(versioned_value::REMOVED_TOKEN)) {
+                slogger.info("restore_replica_count: Detected node {} has left the cluster, status={}, abort restore_replica_count for removing node {}",
+                        endpoint, status, endpoint);
+                if (!as.abort_requested()) {
+                    as.request_abort();
+                }
+                return;
+            }
+            slogger.debug("restore_replica_count: Sleep and detect removing node {}, status={}", endpoint, status);
+            sleep_abortable(std::chrono::seconds(10), as).get();
+        }
+    });
+    auto stop_status_checker = defer([endpoint, &status_checker, &as] () mutable {
+        try {
+            slogger.info("restore_replica_count: Started to stop status checker for removing node {}", endpoint);
+            if (!as.abort_requested()) {
+                as.request_abort();
+            }
+            status_checker.get();
+        } catch (...) {
+            slogger.debug("restore_replica_count: Failed to stop status checker for removing node {}: {}",
```

Comment:
debug level is too weak for unexpected errors.

Reply:
The message is a bit misleading. Abort requesting should not throw, so an exception here really means a failure of status checking, not failure to stop it.

Reply:
The checker is stopped though the abort_source on normal path, I wanted to avoid print the exception on this path. I can ignore the sleep exception separately.

Reply:
Waiting for status_checker is part of the stopping procedure. I think it is fine to say the stopping procedure has failed.

Reply:
> The checker is stopped though the abort_source on normal path, I wanted to avoid print the exception on this path. I can ignore the sleep exception separately.

Yes, we should do that.

Reply:
> Waiting for status_checker is part of the stopping procedure. I think it is fine to say the stopping procedure has failed. Also, if there are multiple steps in the procedure can fail, we do not know which one has failed.

If status_checker failed, then stopping it actually succeeded (it is already stopped). 

It's fine to have a blanket catch, but then exceptions from status_checker.get() should be handled separately.

Reply:
It can be already failed or not yet. 

Anyway, I modified  the log a little bit. See v3.